lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -279,31 +295,38 @@ class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
 
-class OpenSearchPluginRelationClusterNotReadyError(OpenSearchPluginError):
-    """Exception thrown when making API calls and cluster is not yet ready.
-
-    This exception in specific should be handled by classes managing the relations of plugins.
-    """
-
-
 class OpenSearchPluginMissingDepsError(OpenSearchPluginError):
     """Exception thrown when an opensearch plugin misses installed dependencies."""
 
+    def __init__(self, name, deps):
+        self.name = name
+        self.deps = deps
+
 
 class OpenSearchPluginInstallError(OpenSearchPluginError):
     """Exception thrown when opensearch plugin installation fails."""
 
+    def __init__(self, name):
+        self.name = name
```

Comment:
It doesn't seem like we need this constructor?

Reply:
So, I personally do not like to have things undefined. I like well-defined constructs and anyone trying to use these classes will have to follow that pattern. But you're the reviewer, so I will just follow that and remove this definition.